usr/share/rear/build/GNU/Linux/100_copy_as_is.sh

```diff
@@ -168,7 +168,5 @@ Log "LIBS = ${LIBS[@]}"
 
 # Fix ReaR directories when running from checkout or REAR_VAR configuration:
 Log "Validating and fixing ReaR directories for non-default paths"
-test "$VAR_DIR" != /var/lib/rear && ln -v -sf "$VAR_DIR" $ROOTFS_DIR/var/lib/rear
-test "$SHARE_DIR" != /usr/share/rear && ln -v -sf "$SHARE_DIR" $ROOTFS_DIR/usr/share/rear
-
-
+test "$VAR_DIR" != /var/lib/rear && ln -v -srf "$ROOTFS_DIR/$VAR_DIR" $ROOTFS_DIR/var/lib/rear
+test "$SHARE_DIR" != /usr/share/rear && ln -v -srf "$ROOTFS_DIR/$SHARE_DIR" $ROOTFS_DIR/usr/share/rear
```

Comment:
The 'ln' option '-r' is not supported on older systems
e.g. on SLES11 with /bin/ln from coreutils-8.12-6.25.32.33.1

On those systems ReaR can no longer be used from checkout
because then it fails there like
```
2024-12-04 09:13:04.497000665 Validating and fixing ReaR directories for non-default paths
++ test /root/rear-master/var/lib/rear '!=' /var/lib/rear
++ ln -v -srf /var/tmp/rear.C07M5E8PNHq0W1u/rootfs//root/rear-master/var/lib/rear /var/tmp/rear.C07M5E8PNHq0W1u/rootfs/var/lib/rear
ln: invalid option -- 'r'
Try `ln --help' for more information.
++ test /root/rear-master/usr/share/rear '!=' /usr/share/rear
++ ln -v -srf /var/tmp/rear.C07M5E8PNHq0W1u/rootfs//root/rear-master/usr/share/rear /var/tmp/rear.C07M5E8PNHq0W1u/rootfs/usr/share/rear
ln: invalid option -- 'r'
Try `ln --help' for more information.
```
which results later that because of the missing symlinks
the ReaR recovery system startup fails in
usr/share/rear/skel/default/etc/scripts/system-setup
with
```
ERROR: ReaR recovery cannot work without /usr/share/rear/conf/default.conf
```

Because those symbolic links are crucial,
error checking would have helped to show the issue
to the user where it happens, cf.
https://github.com/rear/rear/wiki/Coding-Style#try-hard-to-care-about-possible-errors

Reply:
Ah, interesting. I thought SLES12 is the minimum SLES version we want to support now?

Actually, according to [SUSE Product Support Lifecycle](https://www.suse.com/lifecycle/#suse-linux-enterprise-server-12) SLES12 is already EOL so that we can IMHO also drop support for it in the 2.8 release:
<img width="915" alt="image" src="https://github.com/user-attachments/assets/08b48f96-9c34-4af4-9976-2b3d6f6b0398">

Reply:
SLES12 is already dropped from upstream supported versions, see
https://github.com/rear/rear.github.com/blob/master/documentation/release-notes-2-8.md

The issue here is not what distribution is supported.

The issue is that crucial command exit code was never checked
(also not in the old code before) so it errors out
not during "rear mkrescue" but later during ReaR recovery
system startup with a weird error message that makes it
needlessly hard to find the root cause of the failure.

Reply:
I will provide a fix that errors out here
when symlinking fails.

Reply:
Ah, sorry. Good catch indeed, thanks a lot!

Reply:
Regarding
https://github.com/rear/rear/pull/3206#discussion_r1869015677
"SLES12 is already EOL"
Not really.
Because
https://www.suse.com/lifecycle/#suse-linux-enterprise-server-12
reads
```
Service Pack Release  FCS Date     General Ends LTSS Ends    LTSS Core Ends
...
SUSE Linux Enterprise
Server 12 SP5         09 Dec 2019  31 Oct 2024  31 Oct 2027  October 2030
```
so for SLES12-SP5 LTSS Ends on 31 Oct 2027
(I think 'LTSS Core' does not apply for ReaR)
BUT
ReaR is not supported for "SUSE Linux Enterprise Server 12".
Instead ReaR is supported for
"SUSE Linux Enterprise High Availability Extension 12",
cf. "SUSE support for Relax-and-Recover" in
https://en.opensuse.org/SDB:Disaster_Recovery
so this one actually matters for ReaR
https://www.suse.com/lifecycle/#suse-linux-enterprise-high-availability-extension-12
```
Service Pack Release  FCS Date     General Ends LTSS Ends
...
SUSE Linux Enterprise
High Availability
Extension 12 SP5      09 Dec 2019  31 Oct 2024  31 Oct 2027
```
which shows that 'LTSS Core' does not apply for ReaR and
that ReaR is likely supported up to 31 Oct 2027 by SUSE
provided there will not be some SLE12-SP6
or whatever extended special support 
(one never knows what customers may like to pay for ;-)